native_store.go

```diff
@@ -94,11 +94,11 @@ func (ns *nativeStore) Put(ctx context.Context, serverAddress string, cred auth.
 		dockerCred.Username = emptyUsername
 		dockerCred.Secret = cred.RefreshToken
 	}
-	buffer := new(bytes.Buffer)
-	if err := json.NewEncoder(buffer).Encode(dockerCred); err != nil {
+	credJson, err := json.Marshal(dockerCred)
```

Comment:
nit: golang naming convention
```suggestion
	credJSON, err := json.Marshal(dockerCred)
```